### ISSUE TITLE
Rasa NLU evaluation on a predefined test set

### ISSUE THREAD

**Issue by ahmad-abdellatif** [RasaHQ/rasa#8291]
**Rasa version**: 2.3.0

**Python version**: 3.6.8

**Operating system** (windows, osx, ...): Darwin-20.3.0-x86_64-i386-64bit

**Issue**:
I want to evaluate a trained Rasa nlu model on a predefined test set. To do so, I executed the following command: 
```rasa test nlu -m models/nlu-20210323-152823 -u train_test_split/test_data.yml```

But the command does not generate the intent_report file.

**Error (including full traceback)**:
```
2021-03-24 12:39:15 INFO     rasa.model  - Loading model models/nlu-20210323-152823.tar.gz...
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/rasa/utils/train_utils.py:437: UserWarning: constrain_similarities is set to `False`. It is recommended to set it to `True` when using cross-entropy loss. It will be set to `True` by default, Rasa Open Source 3.0.0 onwards.
  category=UserWarning,
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/rasa/utils/train_utils.py:410: UserWarning: model_confidence is set to `softmax`. It is recommended to set it to `cosine`. It will be set to `cosine` by default, Rasa Open Source 3.0.0 onwards.
  category=UserWarning,
2021-03-24 12:39:23 INFO     rasa.nlu.test  - Running model for predictions:
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 53.57it/s]
2021-03-24 12:39:23 INFO     rasa.nlu.test  - Entity evaluation results:
2021-03-24 12:39:23 INFO     rasa.nlu.test  - Evaluation for entity extractor: DIETClassifier
2021-03-24 12:39:23 WARNING  rasa.test  - No labels to evaluate. Skip evaluation.
2021-03-24 12:39:23 INFO     rasa.nlu.test  - Classification report saved to results/DIETClassifier_report.json.
2021-03-24 12:39:23 INFO     rasa.nlu.test  - Every entity was predicted correctly by the model.
2021-03-24 12:39:23 INFO     rasa.utils.plotting  - Confusion matrix, without normalization:
[[5]]
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/rasa/utils/plotting.py:193: UserWarning: Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.
  axes[0].set(yticks=yticks, xlim=(0, max_xlims[0]), ylim=(min_ylim, max_ylim))
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/rasa/utils/plotting.py:193: UserWarning: Attempting to set identical bottom == top == 0.0 results in singular transformations; automatically expanding.
  axes[0].set(yticks=yticks, xlim=(0, max_xlims[0]), ylim=(min_ylim, max_ylim))
2021-03-24 12:39:24 INFO     rasa.nlu.test  - Evaluation for entity extractor: CRFEntityExtractor
2021-03-24 12:39:24 WARNING  rasa.test  - No labels to evaluate. Skip evaluation.
2021-03-24 12:39:24 INFO     rasa.nlu.test  - Classification report saved to results/CRFEntityExtractor_report.json.
2021-03-24 12:39:24 INFO     rasa.nlu.test  - Incorrect entity predictions saved to results/CRFEntityExtractor_errors.json.
2021-03-24 12:39:25 INFO     rasa.utils.plotting  - Confusion matrix, without normalization:
[[0 0]
 [1 4]]
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/rasa/utils/plotting.py:193: UserWarning: Attempting to set identical left == right == 0 results in singular transformations; automatically expanding.
  axes[0].set(yticks=yticks, xlim=(0, max_xlims[0]), ylim=(min_ylim, max_ylim))
```

**Content of configuration file (config.yml)** (if relevant):
```yml
language: en

pipeline:
# # No configuration for the NLU pipeline was provided. The following default pipeline was used to train your model.
# # If you'd like to customize it, uncomment and adjust the pipeline.
# # See https://rasa.com/docs/rasa/tuning-your-model for more information.
#   - name: WhitespaceTokenizer
#   - name: RegexFeaturizer
#   - name: LexicalSyntacticFeaturizer
#   - name: CountVectorsFeaturizer
#   - name: CountVectorsFeaturizer
#     analyzer: char_wb
#     min_ngram: 1
#     max_ngram: 4
#   - name: DIETClassifier
#     epochs: 100
#   - name: EntitySynonymMapper
#   - name: ResponseSelector
#     epochs: 100
#   - name: FallbackClassifier
#     threshold: 0.3
#     ambiguity_threshold: 0.1

- name: WhitespaceTokenizer
- name: RegexFeaturizer
- name: LexicalSyntacticFeaturizer
- name: CountVectorsFeaturizer
- name: CountVectorsFeaturizer
  analyzer: char_wb
  min_ngram: 1
  max_ngram: 4
- name: CRFEntityExtractor
  features:
  - - pattern
  max_iterations: 50
  L1_c: 0.1
  L2_c: 0.1
- name: DIETClassifier
  epochs: 100
- name: EntitySynonymMapper
- name: DucklingEntityExtractor
  url: http://localhost:8000
  dimensions:
  - time
- name: ResponseSelector
  epochs: 100
- name: FallbackClassifier
  threshold: 0.5
  ambiguity_threshold: 0.1

# Configuration for Rasa Core.
# https://rasa.com/docs/rasa/core/policies/
policies:
# # No configuration for policies was provided. The following default policies were used to train your model.
# # If you'd like to customize them, uncomment and adjust the policies.
# # See https://rasa.com/docs/rasa/policies for more information.
#   - name: MemoizationPolicy
#   - name: TEDPolicy
#     max_history: 5
#     epochs: 100
#   - name: RulePolicy

- name: MemoizationPolicy
- name: TEDPolicy
  max_history: 5
  epochs: 100
- name: RulePolicy
  core_fallback_threshold: 0.5
  core_fallback_action_name: action_default_fallback
```

I noticed that there is a warning in the traceback:
``` 2021-03-24 12:39:24 WARNING  rasa.test  - No labels to evaluate. Skip evaluation. ```

I am not sure how the test file should look like? There is nothing mentioned in the [Rasa documentation](https://rasa.com/docs/rasa/testing-your-assistant#evaluating-an-nlu-model) about that. Did I miss something?

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @m-vdb will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by koaning** [comment]
Just to check, could you share information from your `train_test_split/test_data.yml` file? The confusion matrix suggests there's only 5 examples in it. 

If your primary interest is evaluating your NLU model (emphasis: just the NLU model, ignoring the TED model) then you might prefer to use the cross-validation command. 

```
rasa test nlu --config config.yml \
          --cross-validation --runs 1 --folds 2 \
          --out gridresults
```

Does this help?

**Comment by minarth** [comment]
I have a similar issue as the reporter. I tried the suggested 
```
rasa test nlu --config config.yml \
          --cross-validation --runs 1 --folds 2 \
          --out gridresults
```

I have few intent definition consisting only 1 example. I also have the nlu data split into different files.

According to `-vv` debug, there is intent classification through correct models, but without the results and metrics. Any idea into what should I check? Where can be the issue? thx

**Comment by minarth** [comment]
Can the issue be, that I have empty NLG data (we have moved NLG to a different service and don't have them duplicated inside the original domain.yml)

**Comment by minarth** [comment]
> I have a similar issue as the reporter. I tried the suggested
> 
> ```
> rasa test nlu --config config.yml \
>           --cross-validation --runs 1 --folds 2 \
>           --out gridresults
> ```
> 
> I have few intent definition consisting only 1 example. I also have the nlu data split into different files.
> 
> According to `-vv` debug, there is intent classification through correct models, but without the results and metrics. Any idea into what should I check? Where can be the issue? thx

Nevermind, my custom intent classifier class inherited from `Component` instead of `IntentClassifier`.

**Comment by ahmad-abdellatif** [comment]
> Just to check, could you share information from your `train_test_split/test_data.yml` file? The confusion matrix suggests there's only 5 examples in it.
> 
> If your primary interest is evaluating your NLU model (emphasis: just the NLU model, ignoring the TED model) then you might prefer to use the cross-validation command.
> 
> ```
> rasa test nlu --config config.yml \
>           --cross-validation --runs 1 --folds 2 \
>           --out gridresults
> ```
> 
> Does this help?

Sorry for the late reply. I was able to fix the issue by adding more training examples.
